websites/B/Beatport/presence.ts

```diff
@@ -0,0 +1,122 @@
+import { ActivityType, Assets } from 'premid'
+
+const presence = new Presence({
+  clientId: '1374098754214297670',
+})
+
+let lastPathname = ''
+let startTimestamp = Math.floor(Date.now() / 1000)
+
+presence.on('UpdateData', async () => {
+  const { pathname } = document.location
+  const showTimestamps = false
+
+  if (pathname !== lastPathname) {
+    lastPathname = pathname
+    startTimestamp = Math.floor(Date.now() / 1000)
+  }
+
+  function getReleaseDetails(): string {
+    const song = document.querySelector('.ReleaseDetailCard-style__Name-sc-67b900a1-7.gEsfoH')?.textContent?.trim() || ''
+    const artist = document.querySelector('.ReleaseDetailCard-style__ArtistList-sc-67b900a1-6 a')?.textContent?.trim() || ''
```

Comment:
Consider using selectors which avoid computer-generated class names. You can use [attribute selectors](https://developer.mozilla.org/en-US/docs/Web/CSS/Attribute_selectors) for this:
```ts
[class^=ReleaseDetailCard][class*=Name]
[class^=ReleaseDetailCard][class*=ArtistList] a
```